db/migrate/20220317133810_create_companies.rb

```diff
@@ -3,10 +3,11 @@
 class CreateCompanies < ActiveRecord::Migration[7.0]
   def change
     create_table :companies do |t|
-      t.string :name
-      t.string :symbol
+      t.string :name, null: false
+      t.string :symbol, null: false
 
       t.timestamps
     end
+    add_index :companies, [:name, :symbol], unique: true, name: 'my_index'
```

Comment:
This would make sure there are no 2 companies that have exactly the same: name and symbol (both at the same time). I'm almost sure this is not what you should do. I think you need to ensure there are no 2 companies with exactly the same `symbol`. Also, I guess the name does not have to be unique.

Additionally, I would avoid naming index "my_index". Fortunately, you can just skip `name` attribute and Rails will generated a proper name for you :)

```suggestion
    add_index :companies, :symbol, unique: true
```

Reply:
Ok, I thought that it check them separately, anyway, it is important that symbol is unique,  name should be unique as well,  but if the symbol is checked this is enough.

Reply:
Ok. You may think about how it works for a moment - To ensure uniqueness of two columns separetly, the database would need 2 independed indexes.